ngraph/test/pre_post_process.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/core/pre_post_process/pre_post_process.hpp"
+
+#include <ngraph/pass/visualize_tree.hpp>
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "ngraph/ops.hpp"
+
+using namespace ov;
+using namespace ov::preprocess;
+
+static std::shared_ptr<Function> create_simple_function(element::Type type, const PartialShape& shape) {
+    auto data1 = std::make_shared<ngraph::op::v0::Parameter>(type, shape);
+    data1->set_friendly_name("input1");
+    auto res = std::make_shared<ngraph::op::v0::Result>(data1);
+    res->set_friendly_name("Result");
+    return std::make_shared<ngraph::Function>(ngraph::ResultVector{res}, ngraph::ParameterVector{data1});
+}
+
+TEST(pre_post_process, simple_scale) {
+    auto f = create_simple_function(ngraph::element::f32, ngraph::Shape{1, 3, 4, 5});
+    ngraph::pass::VisualizeTree("before.png").run_on_function(f);  // Visualize the nGraph function to an image
+    f = PrePostProcessor(f)
+            .in()
+            .preprocess()
+            .scale(2.0f)
+            .scale(4.0f)
+            .network()
+            .set_layout(PartialLayout("NCHW"))
+            .build();
```

Comment:
can we create a reference ngraph function and compare with it? 
can we also use different indentation for different sections? clang-format can be turned off such these places

Reply:
yes, sure, currently it is very rough sketch